theano/gof/destroyhandler.py

```diff
@@ -394,6 +394,31 @@ def get_destroyers_of(r):
                 return []
         fgraph.destroyers = get_destroyers_of
 
+        def has_destroyers(protected_list):
+            visited_app_set = set()
+
+            def recursive_destroys_finder(clients_list):
+                for (app, idx) in clients_list:
+                    if app in visited_app_set:
+                        continue
+                    else:
+                        visited_app_set.add(app)
+                    if app == 'output':
+                        continue
+                    destroy_maps = getattr(app.op, 'destroy_map', {}).values()
+                    if idx in [dmap for sublist in destroy_maps for dmap in sublist]:
+                        return True
+                    for var in getattr(app.op, 'view_map', {}).keys():
```

Comment:
that seem good. To help read it, I would rename var to var_idx, to tell it is an index, not a variable.